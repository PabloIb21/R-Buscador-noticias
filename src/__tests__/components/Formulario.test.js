import React from 'react';
import { mount } from 'enzyme';
import Formulario from '../../components/Formulario';

describe('<Formulario />', () => {
  let wrapper;

  beforeEach(() => {
    const guardarCategoria = jest.fn();
    wrapper = mount(<Formulario guardarCategoria={guardarCategoria} />);
  });

  test('mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be able to change the news category', () => {
    const categoria = 'business';
    const select = wrapper.find('select');
    select.simulate('change', { target: { value: categoria } });
    expect(wrapper.find('select').prop('value')).toBe(categoria);
  });
});