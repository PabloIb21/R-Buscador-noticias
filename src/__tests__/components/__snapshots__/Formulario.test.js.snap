// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Formulario /> mount 1`] = `
<div
  className="buscador row"
>
  <div
    className="col s12 m8 offset-m2"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="heading"
      >
        Encuentra noticias por categoria
      </h2>
      <select
        className="browser-default"
        onChange={[Function]}
        value="general"
      >
        <option
          key="general"
          value="general"
        >
          General
        </option>
        <option
          key="business"
          value="business"
        >
          Negocios
        </option>
        <option
          key="entertainment"
          value="entertainment"
        >
          Entretenimiento
        </option>
        <option
          key="health"
          value="health"
        >
          Salud
        </option>
        <option
          key="science"
          value="science"
        >
          Ciencia
        </option>
        <option
          key="sports"
          value="sports"
        >
          Deportes
        </option>
        <option
          key="technology"
          value="technology"
        >
          Tecnolog√≠a
        </option>
      </select>
      <div
        className="input-field col s12"
      >
        <input
          className="btn_block btn-large amber darken-2"
          type="submit"
          value="Buscar"
        />
      </div>
    </form>
  </div>
</div>
`;
